guides/source/asset_pipeline.md

```diff
@@ -8,476 +8,613 @@ This guide covers the asset pipeline.
+helps Rails resolve asset paths at runtime, allowing it to reference the correct
+fingerprinted files.
 
-You can also opt to include controller-specific stylesheets files
-only in their respective controllers using the following:
+The default `manifest.json` includes entries for main assets like
+`application.js` and `application.css` as well as other files, such as images.
+Here's an example of what the JSON might look like:
 
-```html+erb
-<%= stylesheet_link_tag params[:controller] %>
+```json
+{
+  "application.css": "application-6d58c9e6e3b5d4a7c9a8e3.css",
+  "application.js": "application-2d4b9f6c5a7c8e2b8d9e6.js",
+  "logo.png": "logo-f3e8c9b2a6e5d4c8.png",
+  "favicon.ico": "favicon-d6c8e5a9f3b2c7.ico"
+}
 ```
 
-When doing this, ensure you are not using the `require_tree` directive in your `application.css`, as that could result in your controller-specific assets being included more than once.
-
 ### Asset Organization
 
-Pipeline assets can be placed inside an application in one of three locations:
-`app/assets`, `lib/assets` or `vendor/assets`.
-
-* `app/assets` is for assets that are owned by the application, such as custom images or stylesheets.
+Propshaft organizes assets within the `app/assets` directory, which includes
+subdirectories like `images`, `javascripts`, and `stylesheets`. You can place
+your JavaScript, CSS, image files, and other assets into these directories, and
+Propshaft will manage them during the precompilation process.
 
-* `app/javascript` is for your JavaScript code
+You can also add additional asset paths for Propshaft to search by modifying
+`config.assets.paths` in your `config/initializers/assets.rb` file. For example:
 
-* `vendor/[assets|javascript]` is for assets that are owned by outside entities, such as CSS frameworks or JavaScript libraries. Keep in mind that third-party code with references to other files also processed by the asset Pipeline (images, stylesheets, etc.), will need to be rewritten to use helpers like `asset_path`.
-
-Other locations could be configured in the `manifest.js` file, refer to the [Manifest Files and Directives](#manifest-files-and-directives).
-
-#### Search Paths
-
-When a file is referenced from a manifest or a helper, Sprockets searches all of the locations specified in `manifest.js` for it. You can view the search path by inspecting
-[`Rails.application.config.assets.paths`](configuring.html#config-assets-paths) in the Rails console.
+```ruby
+# Add additional assets to the asset load path.
+Rails.application.config.assets.paths << Emoji.images_path
+```
 
-#### Using Index Files as proxies for folders
+Propshaft will make all assets from the configured paths available for serving.
+During the precompilation process, Propshaft copies these assets into the
+`public/assets` directory, ensuring they are ready for production use. This is
+different from Sprockets, which required explicit inclusion of assets in the
```

Comment:
Agree, and I also suggested reordering the sections in a comment above.

Reply:
Thanks you both!  I’ve re-ordered the sections based on your suggestions—thank you! The flow feels much more natural now, and it helped me catch and remove some duplications. As a result, I also combined the comparisons of Propshaft and Sprockets into one section instead of spreading them throughout the guide.